### PR TITLE
Fix more conflicts with SVT-AV1

### PR DESCRIPTION
**Pull request by 1480c1**
Some conflicts somehow still remain.

Closes https://github.com/OpenVisualCloud/SVT-AV1/issues/73

### CODE REVIEW

**Comment by 1480c1** [comment]
Should I do it in this PR?

**Comment by tianjunwork** [comment]
How about refactoring first in one PR follows by this one?

**Comment by AkashiSN** [comment]
Are you planning to merge this PR?
I confirmed that the build is successful by applying this patch.
<details>
<summary>The dockerfile used for confirmation.</summary>

```dockerfile
FROM ubuntu:18.04
ARG MAKEFLAGS="-j7"

ENV PREFIX=/opt/ffmpeg
ENV LD_LIBRARY_PATH="${PREFIX}/lib:${PREFIX}/lib64:/usr/lib64:/usr/lib:/lib64:/lib" \
    PKG_CONFIG_PATH="${PREFIX}/share/pkgconfig:${PREFIX}/lib/pkgconfig:${PREFIX}/lib64/pkgconfig"

RUN apt-get -yqq update && \
    apt-get install -yq --no-install-recommends \
        autoconf \
        automake \
        build-essential \
        ca-certificates \
        curl \
        cmake \
        git \
        pkg-config \
        yasm

## SVT-AV1 https://github.com/OpenVisualCloud/SVT-AV1
RUN DIR=/usr/local/src/svt-av1 && \
    git clone https://github.com/OpenVisualCloud/SVT-AV1.git ${DIR} && \
    cd ${DIR}/ && \
    curl -sLO https://patch-diff.githubusercontent.com/raw/OpenVisualCloud/SVT-AV1/pull/1329.patch && \
    git apply 1329.patch && \
    cd ${DIR}/Build && \
    cmake .. -G"Unix Makefiles" -DCMAKE_INSTALL_PREFIX="${PREFIX}" -DBUILD_SHARED_LIBS=OFF && \
    make ${MAKEFLAGS} install

## SVT-HEVC https://github.com/OpenVisualCloud/SVT-HEVC
RUN DIR=/usr/local/src/svt-hevc && \
    git clone https://github.com/OpenVisualCloud/SVT-HEVC.git ${DIR} && \
    cd ${DIR}/Build && \
    cmake .. -G"Unix Makefiles" -DCMAKE_INSTALL_PREFIX="${PREFIX}" -DBUILD_SHARED_LIBS=OFF && \
    make ${MAKEFLAGS} install

## SVT-VP9 https://github.com/OpenVisualCloud/SVT-VP9
RUN DIR=/usr/local/src/svt-vp9 && \
    git clone https://github.com/OpenVisualCloud/SVT-VP9.git ${DIR} && \
    cd ${DIR} && \
    curl -sLO https://github.com/OpenVisualCloud/SVT-VP9/pull/82.patch && \
    git apply 82.patch && \
    cd ${DIR}/Build && \
    cmake .. -G"Unix Makefiles" -DCMAKE_INSTALL_PREFIX="${PREFIX}" -DBUILD_SHARED_LIBS=OFF && \
    make ${MAKEFLAGS} install

## ffmpeg https://www.ffmpeg.org/
RUN DIR=/usr/local/src/ffmpeg && \
    git clone https://github.com/FFmpeg/FFmpeg ${DIR} && \
    cd ${DIR} && \
    git apply /usr/local/src/svt-hevc/ffmpeg_plugin/0001-lavc-svt_hevc-add-libsvt-hevc-encoder-wrapper.patch && \
    git apply /usr/local/src/svt-av1/ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1-with-svt-hevc.patch && \
    git apply /usr/local/src/svt-vp9/ffmpeg_plugin/master-0001-Add-ability-for-ffmpeg-to-run-svt-vp9-with-hevc-av1.patch && \
    ./configure \
        --disable-shared \
        --enable-libsvtav1 \
        --enable-libsvthevc \
        --enable-libsvtvp9 \
        --enable-static \
        --enable-version3 \
        --extra-cflags="-I${PREFIX}/include" \
        --extra-ldflags="-L${PREFIX}/lib" \
        --pkg-config-flags="--static" \
        --prefix="${PREFIX}" && \
    make ${MAKEFLAGS} && \
    make install
```
</details>

apply this patch

```
$ curl -sLO https://github.com/OpenVisualCloud/SVT-VP9/pull/82.patch
$ git apply 82.patch
```

**Comment by 1480c1** [comment]
forgot that this was stuck in limbo, I'll give it a look if I have time

**Comment by 1480c1** [comment]
@tianjunwork, can we merge this for now and do time and string refactoring later?

**Comment by ImportTaste** [comment]
Was a decision ever made as to whether this would be merged before or after a time and string refactoring?

**Comment by tianjunwork** [comment]
Better to merge it after the time function refactor in #120 so that patch will be cleaner without changes on dead code. If anyone needs it asap, it can be merged first.

File: Source/Lib/ASM_AVX2/EbNoiseExtractAVX2.c [hunk 1]
```diff
@@ -8,11 +8,11 @@
 #include "immintrin.h"
 #include "EbUtility.h"
 
-EB_EXTERN EB_ALIGN(16) const uint8_t filter_type[] = {
+static EB_EXTERN EB_ALIGN(16) const uint8_t filter_type[] = {
```
Comment (tianjunwork) [inline Source/Lib/ASM_AVX2/EbNoiseExtractAVX2.c:11]: filter_type is only used in this file, EB_TXTERN can be removed. Same of weak_chroma_filter

Comment (1480c1) [reply Source/Lib/ASM_AVX2/EbNoiseExtractAVX2.c:11]: going to rebase first to make sure mark's changes doesn't add any new conflicts, then I will remove EB_EXTERN from these

**Comment by 1480c1** [comment]
No more conflicts with the latest changes

File: Source/Lib/ASM_AVX2/EbNoiseExtractAVX2.c [hunk 2]
```diff
@@ -8,16 +8,16 @@
...
 };
 
-EB_EXTERN EB_ALIGN(16) const uint8_t weak_chroma_filter[2][32] = {
+static EB_ALIGN(16) const uint8_t weak_chroma_filter[2][32] = {
         { 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4, 2, 4 },
         { 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2 },
 };
 
-inline void eb_vp9_luma_weak_filter_avx2_intrin(
+static void eb_vp9_luma_weak_filter_avx2_intrin(
```
Comment (tianjunwork) [inline Source/Lib/ASM_AVX2/EbNoiseExtractAVX2.c:20]: `inline` should not be removed. Can we use `static inline`?

Comment (1480c1) [reply Source/Lib/ASM_AVX2/EbNoiseExtractAVX2.c:20]: Sure, I can add inline here

**Pull request merged**